tests/run-make/executable-no-mangle-strip/main.rs

```diff
@@ -0,0 +1,16 @@
+unsafe extern "C" {
+    unsafe static UNDEFINED: usize;
+}
+
+#[unsafe(no_mangle)]
+pub fn used() {
+    println!("UNDEFINED = {}", unsafe { UNDEFINED });
+}
```

Comment:
Honestly, I'm kind of surprised that this worked in the past. Could you describe the use-case?

I'd also be interested, does this pattern work on other platforms?

Reply:
It's used by https://github.com/pgcentralfoundation/pgrx. It's a framework for developing PostgreSQL extensions. This trick is used for writing hybrid (dylib and host tools) code. Code about host tools could be compiled to an executable, since dylib-related code are GC-ed.

This should be a reasonable usage. Please see https://github.com/rust-lang/rust/pull/95604 and https://github.com/rust-lang/rust/pull/95363#issuecomment-1079932354.

Yes. It works for Linux and Windows too.

Reply:
Thanks for the answer. Might make sense to leave a comment in the file about this, and perhaps link to https://github.com/rust-lang/rust/issues/139744, stating that it is a regression test for that?

Reply:
The symbol is present in fhe dynamic library. It's GC-ed iff the final artifact is the executable.

Reply:
An idea then would perhaps be to add another test (or use UI-test revisions) with `#![crate_type = "dylib"]` and `//@ dont-check-compiler-stderr` that expectedly fails?

Reply:
I think it's alreay tested in https://github.com/rust-lang/rust/tree/master/tests/run-make/c-link-to-rust-dylib.

Reply:
No, I meant having test that _fails_ when trying to link undefined symbols in a dylib

Reply:
It is added.

Reply:
Thanks!